modules/apps/transfer/client/cli/tx.go

```diff
@@ -85,24 +86,51 @@ corresponding to the counterparty channel. Any timeout set to 0 is disabled.`),
 			// if the timeouts are not absolute, retrieve latest block height and block timestamp
 			// for the consensus state connected to the destination port/channel
 			if !absoluteTimeouts {
-				consensusState, height, _, err := channelutils.QueryLatestConsensusState(clientCtx, srcPort, srcChannel)
+				clientRes, err := channelutils.QueryChannelClientState(clientCtx, srcPort, srcChannel, false)
 				if err != nil {
 					return err
 				}
 
+				var clientState exported.ClientState
+				if err := clientCtx.InterfaceRegistry.UnpackAny(clientRes.IdentifiedClientState.ClientState, &clientState); err != nil {
+					return err
+				}
+
+				clientHeight, ok := clientState.GetLatestHeight().(clienttypes.Height)
+				if !ok {
+					return fmt.Errorf("invalid height type. expected type: %T, got: %T", clienttypes.Height{}, clientState.GetLatestHeight())
+				}
+
+				var consensusState exported.ConsensusState
+				if clientState.ClientType() != exported.Localhost {
+					consensusStateRes, err := channelutils.QueryChannelConsensusState(clientCtx, srcPort, srcChannel, clientHeight, false)
+					if err != nil {
+						return err
+					}
```

Comment:
it might be faster to just query for the consensus state using the clientRes.ClientID? Otherwise I think another channel query, connection query is performed

Reply:
nice catch, I've updated to make use of `clientutils.QueryConsensusState` using `clientRes.ClientID` as you've suggested